contrib/src/main/java/com/datatorrent/contrib/avro/PojoToAvro.java

```diff
@@ -0,0 +1,264 @@
+      throws NoSuchFieldException, SecurityException
+  {
+    java.lang.reflect.Field f = cls.getDeclaredField(inputFieldName);
+    Class<?> c = ClassUtils.primitiveToWrapper(f.getType());
+
+    Getter<?, ?> classGetter = PojoUtils.createGetter(cls, inputFieldName, c);
+
+    return classGetter;
+  }
+
+  /**
+   * Initializes the list of columns in POJO based on the names from schema
+   */
+  private void initializeColumnMap(Schema schema)
+  {
+    setColumnNames(schema.getFields());
+
+    keyMethodMap = new ArrayList<Getter>();
+    for (int i = 0; i < getColumnNames().size(); i++) {
+      try {
+        keyMethodMap.add(generateGettersForField(cls, getColumnNames().get(i).name()));
+      } catch (NoSuchFieldException | SecurityException e) {
+        throw new RuntimeException("Failed to initialize pojo class getters for field: ", e);
+      }
+    }
+  }
+
+  @InputPortFieldAnnotation(optional = false, schemaRequired = true)
+  public final transient DefaultInputPort<Object> data = new DefaultInputPort<Object>()
+  {
+
+    @Override
+    public void setup(PortContext context)
+    {
+      cls = context.getValue(Context.PortContext.TUPLE_CLASS);
+
+      try {
+        parseSchema();
+        initializeColumnMap(getSchema());
+      } catch (IOException e) {
+        LOG.error("Exception in parsing schema -" + e.getMessage());
+      }
+    }
+
+    @Override
+    public void process(Object tuple)
+    {
+      processTuple(tuple);
+    }
+
+  };
+
+  /**
+   * Converts incoming tuples into Generic records
+   */
+  protected void processTuple(Object tuple)
+  {
+    GenericRecord record = null;
+
+    try {
+      record = getGenericRecord(tuple);
+    } catch (Exception e) {
+      LOG.error("Exception in parsing record");
+      errorCnt++;
+    }
+
+    if (record != null) {
+      output.emit(record);
+      recordCnt++;
+    } else {
+      errorCnt++;
+    }
+  }
+
+  /**
+   * Returns a generic record mapping the POJO fields to provided schema
+   * 
+   * @return Generic Record
+   */
+  private GenericRecord getGenericRecord(Object tuple) throws Exception
+  {
+    int writeErrorCount = 0;
+    GenericRecord rec = new GenericData.Record(getSchema());
+
+    for (int i = 0; i < columnNames.size(); i++) {
+      try {
+        rec.put(columnNames.get(i).name(), AvroRecordHelper.convertValueStringToAvroKeyType(getSchema(),
+            columnNames.get(i).name(), keyMethodMap.get(i).get(tuple).toString()));
+      } catch (AvroRuntimeException e) {
+        LOG.error("Could not set Field [" + columnNames.get(i).name() + "] in the generic record");
+        fieldErrorCnt++;
+      } catch (Exception e) {
+        LOG.error("Parse Exception");
+        fieldErrorCnt++;
+        writeErrorCount++;
+      }
+    }
+
+    if (columnNames.size() == writeErrorCount) {
+      return null;
+    } else {
+      return rec;
+    }
+  }
+
+  @Override
+  public void endWindow()
+  {
+    recordCnt = 0;
```

Comment:
@chandnisingh , Can you please check if this is right place to reset autometrics variable OR should this be done in beginWindow.